Description: When skb_copy_and_csum_datagram_iovec is called for a packet,
for which checksum is already calculated by sw, a harmless message with
backtrace is generated.
Fix: if csum_complete_sw is set, do not dump backtrace message.

From: pchaudhary@linkedin.com
Date: June, 06 2018.

diff --git a/net/core/datagram.c b/net/core/datagram.c
index b46bec3..a7bbe1b 100644
--- a/net/core/datagram.c
+++ b/net/core/datagram.c
@@ -856,7 +856,8 @@ int skb_copy_and_csum_datagram_iovec(struct sk_buff *skb,
 			goto fault;
 		if (csum_fold(csum))
 			goto csum_error;
-		if (unlikely(skb->ip_summed == CHECKSUM_COMPLETE))
+		if (unlikely(skb->ip_summed == CHECKSUM_COMPLETE) &&
+		    !skb->csum_complete_sw)
 			netdev_rx_csum_fault(skb->dev);
 		iov->iov_len -= chunk;
 		iov->iov_base += chunk;
